[BACKLOG-17862] Support for simple literal equality predicates

mkambol opened the pull request:
like

   ( 1 = 0 ) OR ( 1 = 1 ) OR ( 'kitten' = 'dead' )

Added a check in SQLCondition to determine whether the
left and right elements of a condition clause appear to
be literals between an equals function.  If so treat
as literal eq check.

This is an ugly hack around our lack of a real parser.
Some kittens were killed in the making of this commit.

http://jira.pentaho.com/browse/BACKLOG-17862

mkambol committed:
[CLEANUP] checkstyle, minor refactoring, new test

Also removed the 1.6 java source/target from pom.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.jdbc.ThinUtil
 * Branch Change: **-.16%**:small_red_triangle_down:
 * Instruction Change:  + .04%
 * Line Change:  + .04%
 * Method Change:  + .23%

###### org.pentaho.di.core.sql.SQL
 * Branch Change:  + 2.50%
 * Complexity Change:  + 3.64%
 * Instruction Change:  + .70%
 * Line Change:  + 1.65%
 * Method Change:  + 2.86%

###### org.pentaho.di.core.sql.SQLCondition
 * Branch Change:  + 2.56%
 * Complexity Change:  + 3.41%
 * Instruction Change:  + 2.67%
 * Line Change:  + 1.43%
 * Method Change:  + 6.29%

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java. Found 2016. Was expecting 2017

mkambol committed:
[BACKLOG-17862] Support for simple literal equality predicates

like

   ( 1 = 0 ) OR ( 1 = 1 ) OR ( 'kitten' = 'dead' )

Added a check in SQLCondition to determine whether the
left and right elements of a condition clause appear to
be literals between an equals function.  If so treat
as literal eq check.

This is an ugly hack around our lack of a real parser.
Some kittens were killed in the making of this commit.

http://jira.pentaho.com/browse/BACKLOG-17862

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'pdi-dataservice-client' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.jdbc.ThinUtil
 * Branch Change: **-.16%**:small_red_triangle_down:
 * Instruction Change:  + .04%
 * Line Change:  + .04%
 * Method Change:  + .23%

###### org.pentaho.di.core.sql.SQL
 * Branch Change:  + 2.50%
 * Complexity Change:  + 3.64%
 * Instruction Change:  + .70%
 * Line Change:  + 1.65%
 * Method Change:  + 2.86%

###### org.pentaho.di.core.sql.SQLCondition
 * Branch Change:  + 2.56%
 * Complexity Change:  + 3.41%
 * Instruction Change:  + 2.67%
 * Line Change:  + 1.43%
 * Method Change:  + 6.29%

ccaspanello reviewed (approved):
Sad to hear about the cats.   Approved.
